### PR TITLE
Rename "Use the transpile function" section to "Configuring transpilation" and update defaults and configuration options docs

### PR DESCRIPTION
**Pull request by kevinsung**
Part of #885

### CODE REVIEW

**Comment by frankharkins** [comment]
Thanks for converting to a notebook too!

File: docs/transpile/defaults-and-configuration-options.ipynb [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+   "source": [
+    "# Transpilation default settings and configuration options\n",
+    "\n",
+    "Abstract circuits need to be transpiled because systems have a limited set of basis gates and cannot execute arbitrary operations. The transpiler's function is to change arbitrary circuits so that they can run on a specified system.  This is done by translating the circuits to the supported basis gates, and by introducing SWAP gates as needed, so that the circuit's connectivity matches that of the system.\n",
+    "\n",
+    "As explained in [Transpile with pass managers](transpile-with-pass-managers), you can create a [pass manager](/api/qiskit/qiskit.transpiler.PassManager) using the [`generate_preset_pass_manager`](/api/qiskit/transpiler_preset#qiskit.transpiler.preset_passmanagers.generate_preset_pass_manager) function and pass a circuit or list of circuits to its [run](/api/qiskit/qiskit.transpiler.PassManager#run) method to transpile them. You can call `generate_preset_pass_manager` passing only the optimization level and backend, choosing to use the defaults for all other options, or you can pass additional arguments to the function to fine tune the transpilation.\n",
```
Comment (abbycross) [inline docs/transpile/defaults-and-configuration-options.ipynb:12]:
```suggestion
    "As explained in [Transpile with pass managers](transpile-with-pass-managers), you can create a [pass manager](/api/qiskit/qiskit.transpiler.PassManager) using the [`generate_preset_pass_manager`](/api/qiskit/transpiler_preset#qiskit.transpiler.preset_passmanagers.generate_preset_pass_manager) function and pass a circuit or list of circuits to its [run](/api/qiskit/qiskit.transpiler.PassManager#run) method to transpile them. You can call `generate_preset_pass_manager` passing only the optimization level and backend, choosing to use the defaults for all other options, or you can pass additional arguments to the function to fine-tune the transpilation.\n",
```

File: docs/transpile/defaults-and-configuration-options.ipynb [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    "\n",
+    "\n",
+    "All parameters except `optimization_level` are optional.  For full details, see the [Transpiler API documentation](/api/qiskit/transpiler#transpiler-api).\n",
+    "\n",
+    "- `optimization_level` (int) - How much optimization to perform on the circuits. Integer in the range (0 - 3). Higher levels generate more optimized circuits, at the expense of longer transpilation time. See the [Set optimization topic](set-optimization) for more details.\n",
+    "\n",
+    "### Parameters used to describe the compilation target:\n",
+    "\n",
+    "These arguments describe the system that the input circuit will be run on.  For example, they may be used to ensure that the returned circuit can be run on the specified system, where the transpiler maps the circuit's virtual qubits to the physical qubits which may have limited connectivity as specified in the `coupling_map`, `basis_gates` that specify which single- and two-qubit gates are calibrated for the system, and possibly their error rates.\n",
```
Comment (abbycross) [inline docs/transpile/defaults-and-configuration-options.ipynb:114]:
```suggestion
    "### Parameters used to describe the compilation target\n",
    "\n",
    "These arguments describe the system that the input circuit will be run on.  For example, they can be used to ensure that the returned circuit can run on the specified system, where the transpiler maps the circuit's virtual qubits to the physical qubits which may have limited connectivity as specified in the `coupling_map`, `basis_gates` that specify which single- and two-qubit gates are calibrated for the system, and possibly their error rates.\n",
```

Comment (abbycross) [reply docs/transpile/defaults-and-configuration-options.ipynb:114]: Made a couple grammar tweaks but am having trouble parsing the sentence in line 114 (starting with For example) -- possible to rewrite/split up?

File: docs/transpile/defaults-and-configuration-options.ipynb [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+    "- `backend_properties` (BackendProperties) - Properties returned by a system, including information on gate errors, readout errors, qubit coherence times, and so on. Find a system that provides this information by running `backend.properties()`.\n",
+    "- `instruction_durations` (List[Tuple[str, Iterable[int], float, Iterable[float], str]] | List[Tuple[str, Iterable[int], float, Iterable[float]]] | List[Tuple[str, Iterable[int], float, str]] | List[Tuple[str, Iterable[int], float]] | InstructionDurations) - Durations of instructions. Applicable only if `scheduling_method` is specified. The gate lengths defined in `backend.properties` are used by default. They are overwritten if `instruction_durations` is specified. The `instruction_durations` format must be as follows. The instruction_durations must be given as a list of tuples [(instruction_name, qubits, duration, unit), …]. | [(`cx`, [0, 1], 12.3, `ns`), (`u3`, [0], 4.56, `ns`)] | [(`cx`, [0, 1], 1000), (`u3`, [0], 300)] If `unit` is omitted, the default is `dt`, which is a sample time depending on system. If the time unit is `dt`, the duration must be an integer.\n",
```
Comment (abbycross) [inline docs/transpile/defaults-and-configuration-options.ipynb:127]:
```suggestion
    "- `instruction_durations` (List[Tuple[str, Iterable[int], float, Iterable[float], str]] | List[Tuple[str, Iterable[int], float, Iterable[float]]] | List[Tuple[str, Iterable[int], float, str]] | List[Tuple[str, Iterable[int], float]] | InstructionDurations) - Durations of instructions. Applicable only if `scheduling_method` is specified. The gate lengths defined in `backend.properties` are used by default. They are overwritten if `instruction_durations` is specified. The `instruction_durations` format must be as follows. The instruction_durations must be given as a list of tuples [(instruction_name, qubits, duration, unit), …]. | [(`cx`, [0, 1], 12.3, `ns`), (`u3`, [0], 4.56, `ns`)] | [(`cx`, [0, 1], 1000), (`u3`, [0], 300)]. If `unit` is omitted, the default is `dt`, which is a sample time depending on system. If the time unit is `dt`, the duration must be an integer.\n",
```

File: docs/transpile/defaults-and-configuration-options.ipynb [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    "- `timing_constraints` (Dict[str, int] | None) - An optional control hardware restriction on instruction time resolution. This information is provided by the system configuration. If the system doesn’t have any restriction on the instruction time allocation, `timing_constraints` is `None` and no adjustment is performed. A system might report a set of restrictions, namely:\n",
+    "    - `granularity`: An integer value representing the minimum pulse gate resolution in units of dt. A user-defined pulse gate should have duration that is a multiple of this granularity value.\n",
+    "    - `min_length`: An integer value representing the minimum pulse gate length in units of dt. A user-defined pulse gate should be longer than this length.\n",
+    "    - `pulse_alignment`: An integer value representing a time resolution of gate instruction starting time. Gate instructions should start at time that is a multiple of this value.\n",
+    "    - `acquire_alignment`: An integer value representing a time resolution of measure instruction starting time. Measure instruction should start at time that is a multiple of this value.\n",
```
Comment (abbycross) [inline docs/transpile/defaults-and-configuration-options.ipynb:132]:
```suggestion
    "- `timing_constraints` (Dict[str, int] | None) - An optional control hardware restriction on instruction time resolution. This information is provided by the system configuration. If the system doesn’t have any restriction on the instruction time allocation, `timing_constraints` is `None` and no adjustment is performed. A system might report a set of restrictions, namely:\n",
    "    - `granularity`: An integer value representing the minimum pulse gate resolution in units of dt. A user-defined pulse gate should have a duration that is a multiple of this granularity value.\n",
    "    - `min_length`: An integer value representing the minimum pulse gate length in units of dt. A user-defined pulse gate should be longer than this length.\n",
    "    - `pulse_alignment`: An integer value representing a time resolution of gate instruction starting time. Gate instructions should start at a time that is a multiple of this value.\n",
    "    - `acquire_alignment`: An integer value representing a time resolution of measure instruction starting time. Measure instruction should start at a time that is a multiple of this value.\n",
```

File: docs/transpile/defaults-and-configuration-options.ipynb [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+    "\n",
+    "<details>\n",
+    "  <summary>\n",
+    "    **Layout and topology parameters**\n",
+    "  </summary>\n",
+    "\n",
+    "- `basis_gates` (List[str] | None) - List of basis gate names to unroll to.  For example ['u1', 'u2', 'u3', 'cx']. If `None`, do not unroll.\n",
+    "- `coupling_map` (CouplingMap | List[List[int]]) - Directed coupling map (possibly custom) to target in mapping. If the coupling map is symmetric, both directions need to be specified. These formats are supported:\n",
+    "    - CouplingMap instance\n",
+    "    - List, must be given as an adjacency matrix, where each entry specifies all directed two-qubit interactions supported by the system.  For example:[[0, 1], [0, 3], [1, 2], [1, 5], [2, 5], [4, 1], [5, 3]]\n",
```
Comment (abbycross) [inline docs/transpile/defaults-and-configuration-options.ipynb:143]:
```suggestion
    "    - List - must be given as an adjacency matrix, where each entry specifies all directed two-qubit interactions supported by the system.  For example: [[0, 1], [0, 3], [1, 2], [1, 5], [2, 5], [4, 1], [5, 3]]\n",
```

File: docs/transpile/defaults-and-configuration-options.ipynb [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+    "- `inst_map` (List[InstructionScheduleMap] | None) - Mapping of circuit operations to pulse schedules. If `None`, the system’s `instruction_schedule_map` is used.\n",
+    "</details>\n",
+    "\n",
+    "\n",
+    "### Parameters used to influence how the transpiler works:\n",
```
Comment (abbycross) [inline docs/transpile/defaults-and-configuration-options.ipynb:148]:
```suggestion
    "### Parameters used to influence how the transpiler works\n",
```

File: docs/transpile/defaults-and-configuration-options.ipynb [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "  <summary>\n",
+    "    **Scheduling stage**\n",
+    "  </summary>\n",
+    "\n",
+    "- `scheduling_method` (str) - Name of the scheduling pass. This can also be the external plugin name to use for the scheduling stage. You can see a list of installed plugins by running `list_stage_plugins()` with `scheduling` for the `stage_name` argument.\n",
+    "  * 'as_soon_as_possible': Schedule instructions greedily, as early as possible on a qubit resource. (alias: `asap`)\n",
+    "  * 'as_late_as_possible': Schedule instructions late, that is, keeping qubits in the ground state when possible. (alias: `alap`).  This is the default.\n",
```
Comment (abbycross) [inline docs/transpile/defaults-and-configuration-options.ipynb:204]:
```suggestion
    "  * 'as_soon_as_possible': Schedule instructions greedily, as early as possible on a qubit resource (alias: `asap`).\n",
    "  * 'as_late_as_possible': Schedule instructions late, that is, keeping qubits in the ground state when possible (alias: `alap`).  This is the default.\n",
```

**Review by abbycross — APPROVED** [review]
Great work, thank you! Just a few teensy things in the comments

File: docs/transpile/defaults-and-configuration-options.ipynb (hunk 4 above)
Comment (kevinsung) [reply docs/transpile/defaults-and-configuration-options.ipynb:132]: Thanks, it's an improvement. I think these were copied straight from the API docs, so the issue persists there.

File: docs/transpile/defaults-and-configuration-options.ipynb (hunk 2 above)
Comment (kevinsung) [reply docs/transpile/defaults-and-configuration-options.ipynb]: I rewrote the sentence
> These arguments describe the target system for circuit execution, including information such as the coupling map of the system (which describes the connectivity of the qubits), the basis gates supported by the system, and even the error rates of the gates.

**Pull request merged**